tools/update_dtb_initrd.py

```diff
@@ -25,10 +25,19 @@ def update_node_property(node, property_name, property_value):
 
 
 def update_chosen(node, initrd_start, initrd_size):
+    # DTS Protocol dictates >32-bit values be represented in pairs. ie, 0x800000000 -> 0x8 0x0
+    initrd_end = initrd_start + initrd_size
+    if initrd_start > 0xFFFFFFFF:
+        update_start = [initrd_start >> 32, initrd_start & 0xFFFFFFFF]
+        update_end = [initrd_end >> 32, initrd_end & 0xFFFFFFFF]
+    else:
+        update_start = [initrd_start]
+        update_end = [initrd_end]
```

Comment:
This does not handle the case where `initrd_start` is below the 32-bit border and `update_end` is after it. Might be somethign that rarely happens. 

```
        update_start = [initrd_start] if initrd_start <= 0xFFFFFFFF \
                       else [initrd_start >> 32, initrd_start & 0xFFFFFFFF]

        update_end = [initrd_end] if initrd_end <= 0xFFFFFFFF \
                       else [initrd_end >> 32, initrd_end & 0xFFFFFFFF]
```

I wonder, would the most simple handling be, that we that we always use 2 values. Or would this break 32-bit platforms then?

Reply:
I agree with @axel-h's first suggestion.

Reply:
Nice catch, Axel. I've implemented that change. I also think its a very rare edge case, but its good to have a check for it.